code/implementation/settings-api/src/main/java/io/cattle/platform/api/settings/manager/SettingsFilter.java

```diff
@@ -14,11 +15,14 @@
 
 import org.apache.commons.collections.Predicate;
 
+import com.netflix.config.DynamicStringListProperty;
+
 public class SettingsFilter implements Predicate {
 
     Set<String> publicSettings;
     boolean all;
     boolean canListAll;
+    public static final DynamicStringListProperty SETTINGS_HIDDEN = ArchaiusUtil.getList("settings.hidden");
```

Comment:
We have no ability to really create a hard coded setting so a setting to hide settings just creates a security problem.  I would opt for a change in design in that you can create a setting that has a hidden=true flag on it.  Then we can properly encrypt the values.  If that is too much up front effort and you want to do a two phased approach, then start here with a hardcoded list of properties to hide.

A third option is to have a regexp setting of what to hide, like `.*password.*|.*secret.*`.  If it was one simple static setting then we could put it in the `Dockerfile` of `rancher/server` and it would hide the secret.

@cjellick